data/data.go

```diff
@@ -31,6 +31,70 @@ type EffxYaml struct {
 	FilePath string
 }
 
+// given a absolute path (example: /runner/folder/root/effx.yaml)
+// and given a workingDir (example: root)
+// it will return root/effx.yaml
+func parseRelativePathFromAbsolutePath(absoluteDir, workingDir string) string {
+	res := strings.Split(absoluteDir, workingDir)
+	if len(res) > 1 {
+		contentPath := res[1]
+		return workingDir + contentPath
+	}
+	return absoluteDir
+}
+
+// given an absolute path, this function will find the root directory
+// (directory with a .git file)
+func findGitRootDirectory(absoluteDir string) (string, error) {
+	pathDir := filepath.Dir(absoluteDir)
+
+	for pathDir != "" {
+		pathDir = filepath.Clean(pathDir)
+		files, err := ioutil.ReadDir(pathDir)
+
+		if err != nil {
+			return "", err
+		}
+
+		for _, file := range files {
+			if !file.IsDir() {
+				if strings.Contains(file.Name(), ".git") {
+					return filepath.Base(pathDir), nil
+				}
+			}
+		}
+
+		pathDir = filepath.Join(pathDir, "..")
+	}
+
+	return absoluteDir, nil
+}
+
+// converts /runner/sandbox/root/effx.yaml
+// to root/effx.yaml for a more helpful file source.
+// it will try to get working directory from circleci,
+// if cannot find it, it will attempt to find the root directory
+// where git files are present.
+func convertToRelativePath(absoluteDir string) (string, error) {
```

Comment:
I think it might be easier to extend this logic to other CI systems if all we needed was the repository name.

From different sources:
[CircleCI](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables): `CIRCLE_PROJECT_REPONAME`
[Gitlab](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html): `CI_PROJECT_NAME`
[Github](https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables): `GITHUB_REPOSITORY`

Reply:
Another option is to use: `git rev-parse --show-toplevel`

I think with this approach, it'd be worth embedding `go-git` to ensure the CLI doesn't have any missing dependencies: https://git-scm.com/book/en/v2/Appendix-B%3A-Embedding-Git-in-your-Applications-go-git

Reply:
+1 to making this easier to extend. We already use `go-git` in vcs-connect but adding it here to support detecting repositories would be pretty easy to do. Also:

- [Semaphore CI](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#git-repository-related): `SEMAPHORE_GIT_URL`
- [Jenkins][]: ???

Reply:
as far as the git library, we can start using it, but the command you mentioned is not supported https://github.com/go-git/go-git/issues/74

Reply:
I can add the semaphore env